lib/driip-settlement.js

```diff
@@ -174,17 +194,20 @@ class DriipSettlement {
         const settled = await this.hasPaymentDriipSettled(nonce, address);
         const latestNonce = await this.getCurrentProposalNonce(address, currency.ct, currency.id);
```

Comment:
Run all in parallel using "Promise.all" ?

Reply:
+ catch and rethrow if anyone of these fails

Reply:
Updated using promise.all and rethrow using NestedError